code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -285,6 +300,17 @@
 		playsound(loc, src.attack_sound, 50, 1, 1)
 		playsound(loc, src.attack_sound, 50, 1, 1)
 		playsound(loc, src.attack_sound, 50, 1, 1)
+	if(istype(target, /turf/simulated/wall/r_wall))
+		if(prob(10))
+			playsound(src, 'sound/effects/meteorimpact.ogg', 100, 1)
+			src << text("<span class='notice'>You smash through the wall.</span>")
+			src.say("[src.battlecry][src.battlecry][src.battlecry][src.battlecry][src.battlecry][src.battlecry][src.battlecry][src.battlecry][src.battlecry][src.battlecry]\
```

Comment:
I think you can move the say to outside of the if..else because it'll happen either way. And what about making the meteor noise in both cases, much like a hulk smashing a wall?